tests/unit/test_utils.py

```diff
@@ -297,19 +299,25 @@ def mode(self, path):
     def confirm_files(self):
         # expectations based on 022 umask set above and the unpack logic that
         # sets execute permissions, not preservation
-        for fname, expected_mode, test in [
-                ('file.txt', 0o644, os.path.isfile),
-                ('symlink.txt', 0o644, os.path.isfile),
-                ('script_owner.sh', 0o755, os.path.isfile),
-                ('script_group.sh', 0o755, os.path.isfile),
-                ('script_world.sh', 0o755, os.path.isfile),
-                ('dir', 0o755, os.path.isdir),
-                (os.path.join('dir', 'dirfile'), 0o644, os.path.isfile)]:
+        for fname, expected_mode, test, expected_contents in [
+                ('file.txt', 0o644, os.path.isfile, 'file\n'),
+                # We don't test the "symlink.txt" contents for now.
+                ('symlink.txt', 0o644, os.path.isfile, None),
+                ('script_owner.sh', 0o755, os.path.isfile, 'file\n'),
+                ('script_group.sh', 0o755, os.path.isfile, 'file\n'),
+                ('script_world.sh', 0o755, os.path.isfile, 'file\n'),
+                ('dir', 0o755, os.path.isdir, None),
+                (os.path.join('dir', 'dirfile'), 0o644, os.path.isfile, ''),
+        ]:
             path = os.path.join(self.tempdir, fname)
             if path.endswith('symlink.txt') and sys.platform == 'win32':
                 # no symlinks created on windows
                 continue
             assert test(path), path
+            if expected_contents is not None:
+                with open(path) as f:
```

Comment:
Maybe explicitly set an encoding here? Or read (and make the expected value) bytes? The expected data is all ASCII, so it shouldn't in theory matter, but explicit is better than implicit and all that...

Reply:
Thanks, @pfmoore. Good point. Originally I thought about making it bytes, but then I saw the tests were passing anyways, so I didn't think I needed to. :) I'll go ahead and change it because I agree the definiteness is better.